test/filestorage.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { create } = require('..');
+const metatests = require('metatests');
+const common = require('metarhia-common');
+
+const testDir = path.join(__dirname, 'fs-root');
+const fsTest = metatests.test('Filestorage');
+
+const minCompressSize = 1000;
+
+fsTest.test('Write string to file', test => {
+  const dir = path.join(testDir, '0');
+  create({ dir, minCompressSize }, (err, storage) => {
+    test.error(err);
+    storage.write(new common.Uint64(0), 'data0',
+      { checksum: 'CRC32', dedupHash: 'SHA256' },
+      (err, checksum, dedupHash, originalSize) => {
+        test.error(err);
+        test.strictSame(checksum, '20cd1c30');
+        test.strictSame(dedupHash,
+          '4285d10832a8edf4cc7979e9a257e145dc5c934549f62e0bf1dc6070e67cc4ab');
+        test.strictSame(originalSize, 5);
+
+        fs.readFile(path.join(dir, '0000', '0000.f'), (err, data) => {
```

Comment:
Could you please use FS_EXT in tests instead of manually writing it down.
Also I think you should have 'z' (complressed files prefix) be extracted too.